src/features/drop-manager/routes/nft/[id].tsx

```diff
@@ -1,71 +1,137 @@
-      justifyContent: 'right',
-      verticalAlign: 'middle',
-    },
-    loadingElement: <Skeleton height="30px" />,
-  },
-];
-
-const data: NFTDropResponse = {
-  name: 'Star Invader 3',
-  links: [
-    { id: 1, slug: '#2138h823h', hasClaimed: true },
-    { id: 2, slug: '#2138h823h', hasClaimed: false },
-    { id: 3, slug: '#c34fd2n32', hasClaimed: false },
-    { id: 4, slug: '#rf5hhfaxm', hasClaimed: true },
-  ],
-};
+import { tableColumns } from '../../components/TableColumn';
+import { INITIAL_SAMPLE_DATA } from '../../constants/common';
+import { setConfirmationModalHelper } from '../../components/ConfirmationModal';
 
 export default function NFTDropManagerPage() {
+  const { setAppModal } = useAppContext();
+  const toast = useToast();
+
+  const { id: dropId } = useParams();
   const [loading, setLoading] = useState(true);
 
-  // TODO: Remove this after backend is ready
+  const [name, setName] = useState('Drop');
+  const [dataSize, setDataSize] = useState<number>(0);
+  const [data, setData] = useState<DataItem[]>([INITIAL_SAMPLE_DATA[0]]);
+
+  const [wallet, setWallet] = useState({});
+  const { selector, accountId } = useAuthWalletContext();
+
   useEffect(() => {
-    setTimeout(() => {
-      setLoading(false);
-    }, 5000);
+    const getWallet = async () => {
+      setWallet(await selector.wallet());
+    };
+    getWallet();
   }, []);
 
-  // TODO: consider moving these to DropManager if backend request are the same for NFT and Ticket
-  const handleCopyClick = () => {
-    // TODO: copy handler
+  const {
+    hasPagination,
+    pagination,
+    firstPage,
+    lastPage,
+    loading: paginationLoading,
+    handleNextPage,
+    handlePrevPage,
+  } = usePagination({
+    dataSize,
+    handlePrevApiCall: async () => {
+      await handleGetDrops({
+        pageIndex: pagination.pageIndex - 1,
+        pageSize: pagination.pageSize,
+      });
+    },
+    handleNextApiCall: async () => {
+      await handleGetDrops({
+        pageIndex: pagination.pageIndex + 1,
+        pageSize: pagination.pageSize,
+      });
+    },
+  });
+
+  const handleGetDrops = async ({ pageIndex = 0, pageSize = PAGE_SIZE_LIMIT }) => {
+    if (!accountId) return;
+    let drop = await getDropInformation({
+      dropId,
+    });
+    if (!drop)
+      drop = {
+        metadata: '{}',
+      };
+
+    setDataSize(drop.next_key_id);
+
+    setName(JSON.parse(drop.metadata as unknown as string).dropName);
+
+    const { publicKeys, secretKeys } = await generateKeys({
+      numKeys: Math.min(drop.next_key_id, pageSize),
+      rootEntropy: `${get(MASTER_KEY) as string}-${dropId}`,
+      autoMetaNonceStart: pageIndex * pageSize,
+    });
+
+    const keyInfo = await getKeyInformationBatch({
+      publicKeys,
+      secretKeys,
+    });
+
+    setData(
+      secretKeys.map((key, i) => ({
+        id: i,
+        publicKey: publicKeys[i],
+        link: `https://keypom.xyz/claim/${dropId}#${key.replace('ed25519:', '')})`,
```

Comment:
@andrehadianto instead of `dropId`, we should be using `contractId`. You can get `contractId` from sdk's `getEnv()`.
```
const { contractId } = getEnv();